docs: README advertises frontend test commands that do not exist

## Summary
The README tells contributors to run `pnpm test` and `pnpm test:watch`, but those scripts are not defined in `package.json`.

## Evidence
- The README documents `pnpm test` and `pnpm test:watch` in [README.md](README.md) lines 160-170.
- `package.json` only defines `dev`, `build`, `preview`, `lint`, `lint:fix`, `format`, `format:check`, `typecheck`, and `tauri` in [package.json](package.json) lines 6-15.
- Running `pnpm run test` currently fails with `ERR_PNPM_NO_SCRIPT Missing script: test`.

## Impact
New contributors will follow the documented workflow and immediately hit a dead end.

## Suggested fix
- Either add the documented Vitest scripts, or remove them from the README and CI expectations.
- Keep the README command matrix generated or validated from `package.json` so it cannot drift again.

--- a/package.json
+++ b/package.json
@@ -12,6 +12,9 @@
     "format": "prettier --write \"src/**/*.{ts,tsx,css}\"",
     "format:check": "prettier --check \"src/**/*.{ts,tsx,css}\"",
     "typecheck": "tsc --noEmit",
+    "test": "vitest run",
+    "test:watch": "vitest",
+    "test:coverage": "vitest run --coverage",
     "tauri": "tauri"
   },
   "dependencies": {
@@ -58,6 +61,9 @@
     "prettier": "^3.0.0",
     "tailwindcss": "^4.2.2",
     "typescript": "^5.6.0",
-    "vite": "^6.4.2"
+    "@vitest/coverage-v8": "^3.0.0",
+    "jsdom": "^26.0.0",
+    "vite": "^6.4.2",
+    "vitest": "^3.0.0"
   }
 }

